nservicebus/upgrades/5to6-migrate-existing-handlers.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Migrate existing handlers/sagas to Version 6
+summary: Describes how to migrate existing handlers/sagas to Version 6
+---
+
+In previous versions of NServiceBus, a typical handler class looks like the below:
+
+snippet:5to6-handler-migration-beginning
+
+Implement the new `IHandleMessages` interface. Remove any code that is generated by the IDE or additional tools like Resharper and mark the new `Handle` method as `async` by adding the `async` keyword.
+
+snippet:5to6-handler-migration-step1
+
+Rename the bus property or the bus constructor parameter to `context`.
+
+snippet:5to6-handler-migration-step2
+
+Inline or "cut-and-paste" the old `Handle` method code into the new asynchronous `Handle` method. When the code is compiled a compiler warning [`CS4014`](https://msdn.microsoft.com/en-us/library/hh873131.aspx) will be shown indicating that there are asynchronous methods in the handler which are not awaited.
+
+snippet:5to6-handler-migration-step3
+
+Fix the compiler warnings by introducing the `await` statement followed by `ConfigureAwait(false)` to each asynchronous method call.
+
+INFO: Visual Studio Versions which support Roslyn have the capability to automatically fix those warnings with the `Ctrl+.` (depending on the keybindings) shortcut. It is even possible to fix it in the whole solution if desired.
```

Comment:
i dont think people care/know about "supports roslyn". perhaps just state "after version x of VS"